### PR TITLE
Compute: update 2019 07 01

### PR DESCRIPTION
**Pull request by ChenTanyi**
#903 
@yaohaizh here is the test:
for VM, maxPrice cannot be updated when VM is in deallocated state, priority can be update between LOW and SPOT, but cannot update between REGULAR and others.
for VMSS, maxPrice can be updated any time, but the priority cannot be updated, since swagger doesn't have such field. https://github.com/Azure/azure-rest-api-specs/blob/f57417493691c0d9a15440ffd7fdb4173bf7295c/specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json#L9200

### CODE REVIEW

File: azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java [hunk 1]
```diff
@@ -207,12 +207,49 @@ public void canCreateLowPriorityVirtualMachine() throws Exception {
...
+        try {
+            foundVM.update()
+                    .withMaxPrice(1500.0)
+                    .apply();
+            // not run to assert
+            Assert.assertEquals((Double) 1500.0, foundVM.billingProfile().maxPrice());
+            Assert.assertTrue(false);
+        } catch (AssertionError e) {
+            throw e;
+        } catch (Exception e) {} // cannot change max price when vm is running
```
Comment (weidongxu-microsoft) [inline azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java]: Maybe better to catch CloudException (and then there would be no need to catch-rethrow AssertionError)? No need to record again.

File: azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java [hunk 2]
```diff
@@ -207,12 +207,49 @@ public void canCreateLowPriorityVirtualMachine() throws Exception {
...
+        try {
+            foundVM.update()
+                    .withPriority(VirtualMachinePriorityTypes.REGULAR)
+                    .apply();
+            // not run to assert
+            Assert.assertEquals(VirtualMachinePriorityTypes.REGULAR, foundVM.priority());
+            Assert.assertTrue(false);
+        } catch (AssertionError e) {
+            throw e;
+        } catch (Exception e) {} // cannot change priority from spot to regular
```
Comment (weidongxu-microsoft) [inline azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java]: Just asking. Is there any service message saying that cannot be done and provide reason?

Comment (ChenTanyi) [reply azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java]: No, the exception is just: `com.microsoft.azure.CloudException: Changing property 'priority' is not allowed.` These contexts are all from services email.

**Comment by ChenTanyi** [comment]
disk import data error due to service 400, cannot be merged

File: azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java [hunk 3]
```diff
@@ -244,7 +244,7 @@ public void canCreateUpdatePriorityAndPrice() throws Exception {
             // not run to assert
             Assert.assertEquals(VirtualMachinePriorityTypes.REGULAR, foundVM.priority());
             Assert.fail();
-        } catch (CloudException e) {} // cannot change priority from spot to regular
+        } catch (CloudException e) {} // cannot change priority from low to regular
```
Comment (yaohaizh) [inline azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java]: Please confirm this behavior with compute service team

Comment (ChenTanyi) [reply azure-mgmt-compute/src/test/java/com/microsoft/azure/management/compute/VirtualMachineOperationsTests.java]: This is written by service team, and my test also fits their statement.

**Pull request merged**